### PR TITLE
feat(replay): Implement the View in Fullscreen button

### PR DESCRIPTION
**Pull request by ryan953**
Implement the 'View in full screen' button that shows the player, as well as the play/pause controls.

Screen Shot 2022-04-12 at 3 36 18 PM

### CODE REVIEW

**Comment by ryan953** [comment]
Current dependencies on/for this PR:
* master
  * **PR #33566** <a href="https://app.graphite.dev/github/pr/getsentry/sentry/33566" target="_blank">Graphite</a>  (this PR)

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/getsentry/sentry/33566?utm_source=stack-comment).

File: package.json [hunk 1]
```diff
@@ -132,6 +132,7 @@
     "regenerator-runtime": "^0.13.3",
     "rrweb": "^1.1.3",
     "rrweb-player": "^0.7.13",
+    "screenfull": "^6.0.1",
```
Comment (ryan953) [inline package.json:135]: New dep: MIT license. 

It's about 150 lines if we wanted to copy+paste it instead

https://www.npmjs.com/package/screenfull
https://github.com/sindresorhus/screenfull

File: static/app/components/replays/replayController.tsx [hunk 2]
```diff
@@ -115,17 +118,23 @@ const ReplayControls = ({
...
+          onClick={isFullscreen ? exitFullscreen : onFullscreen}
         />
       </ButtonGrid>
-    </Column>
+    </React.Fragment>
   );
 };
 
-const Column = styled('div')`
+const Panel = styled(_Panel)<{isFullscreen: boolean}>`
```
Comment (billyvg) [inline static/app/components/replays/replayController.tsx:132]: I would avoid using `_` prefix in this way as it has a special meaning w/ eslint to denote as unused (e.g. when destructuring or an arg that is not used). Maybe `Base-` as prefix?

File: static/app/components/replays/useFullscreen.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    return () => screenfull.off('change', onChange);
+  });
+
+  return {
+    isEnabled: screenfull.isEnabled,
+    ref,
+    isFullscreen,
+    enter,
+    exit,
+  };
```
Comment (billyvg) [inline static/app/components/replays/useFullscreen.tsx:38]: Can we add some docs for each of these? 

If we use the multiline comments here, will TS pick them up like they do for types? if not we may want to explicitly define a return type and comment that instead.

File: static/app/views/replays/details.tsx [hunk 4]
```diff
@@ -106,9 +107,21 @@ function getProjectSlug(event: Event) {
   return event.projectSlug || event['project.name']; // seems janky
 }
 
+// TODO: investigate the `:fullscreen` CSS selector
```
Comment (billyvg) [inline static/app/views/replays/details.tsx:110]: I would add a scope to your TODO, to make it easier to ~ignore~ grep for. e.g. ` TODO(replay)`

File: static/app/components/replays/replayContext.tsx [hunk 5]
```diff
@@ -127,7 +121,6 @@ export function Provider({children, events}: React.PropsWithChildren<Props>) {
     if (replayerRef.current && events) {
       initRoot(replayerRef.current.wrapper.parentElement as RootElem);
     }
-    return cleanupListeners;
```
Comment (ryan953) [inline static/app/components/replays/replayContext.tsx:130]: This was running too often, causing problems.

File: static/app/components/replays/replayController.tsx [hunk 6]
```diff
@@ -115,17 +118,23 @@ const ReplayControls = ({
...
           data-test-id="replay-fullscreen"
           size="xsmall"
-          title={t('View in full screen')}
+          title={isFullscreen ? t('Exit full screen') : t('View in full screen')}
+          aria-label={isFullscreen ? t('Exit full screen') : t('View in full screen')}
           icon={<IconResize size="sm" />}
-          onClick={onFullscreen}
-          aria-label={t('View in full screen')}
+          priority={isFullscreen ? 'primary' : undefined}
+          onClick={isFullscreen ? exitFullscreen : onFullscreen}
```
Comment (billyvg) [inline static/app/components/replays/replayController.tsx:125]: Not sure what `onFullscreen` is used for, but do we want to make the callbacks more consistent (e.g. should we have `onExitFullscreen`? or `onFullscreenToggle: (isFullscreen: bool) => void`

Read down further and yeah seems like this is a bit inconsistent?

**Review by billyvg — APPROVED** [review]
Let's merge this and can iterate on the callback if needed

**Comment by billyvg** [comment]
@ryan953 oh hmm, let's deal with the vercel thing first

**Pull request merged**